crates/stc_ts_types/src/lib.rs

```diff
@@ -1062,6 +1062,18 @@ impl TypeElement {
             TypeElement::Index(_) => None,
         }
     }
+
+    pub fn get_type(&self) -> Option<Type> {
```

Comment:
Please change return type to `Option<&Type>`

Reply:
```rs
let ty: &Option<Box<Type>> = match self {
            TypeElement::Call(CallSignature { ret_ty, .. }) => ret_ty,
            TypeElement::Constructor(ConstructorSignature { ret_ty, .. }) => ret_ty,
            TypeElement::Property(PropertySignature { type_ann, .. }) => type_ann,
            TypeElement::Method(MethodSignature { ret_ty, .. }) => ret_ty,
            TypeElement::Index(IndexSignature { type_ann, .. }) => type_ann,
        };

        ty.as_ref().map(|ty: &Box<Type>| *ty.clone())
```
umm... Could you reconsider?